### ISSUE TITLE
rainfall data size of data-length mismatch

### ISSUE THREAD

**Issue by jeevakir** [iamsaswata/imdlib#22]
Exception: Error in file reading,mismatch in size of data-length occurs when handling rainfall data for any year.  I have updated the library to 0.1.15 and the issue still persisted. I got around the problem with commenting out the lines 322 and 323 in core.py

https://github.com/iamsaswata/imdlib/blob/148ce108fe366454e08afb1302c68ee2a8ca9ab6/imdlib/core.py#L318-L323

am i making some stupid mistake or is it problem with the imd data being changed to old format that makes the new code reduntant?

**Comment by iamsaswata** [comment]
Hi @jeevakir. I found some changes in the precip data from the new IMD server (https://imdpune.gov.in/cmpg/Griddata/rainfall.php) over the old ones (https://imdpune.gov.in/Clim_Pred_LRF_New/rainfall.php). For tackling these changes, Line 322 and 323 are added. So, it means you can not use your old downloaded data with new IMDLIB (>0.1.14) package. I guess if you use newer (>=0.1.15) IMDLIB, please download the data again and see if that works. Another possibility is to you use earlier downloaded data with IMDLIB <=0.1.13. Curious to know, if that works for you.

**Comment by jeevakir** [comment]
Hi @iamsaswata, sorry for the abrupt questions without greetings. The strange thing is i deleted all the GRD files and then updated imdlib to 0.1.15 using git. it was already in the version 0.1.14. I also ensured whether it is downloading from the newer URL.  I posted the issue after trying these out and getting the same error. i don't have any old data with me. it seems strange to me as well. i didn't have chance to check it for the past few months. i already searched the issues and i knew about the changes, that's how i was able to locate the issue in my setup. if its not too much of trouble can you check the source data once again?.

**Comment by iamsaswata** [comment]
Thanks @jeevakir for bringing this up. I checked and found the same error as you reported. Looks like there are some changes in the data again (basically now the new data from the new server is same as the data from the old server), which doesn't require the extra code block I created to handle the data from the new server. So, I commented out this section #L320-L324.
https://github.com/iamsaswata/imdlib/blob/37e435e082bed43103d6096a585bc9a9c94f2fd4/imdlib/core.py#L320-L324

With this slight changes, looks like it is working fine again. I have pushed a new update (0.1.16) for this. Hopefully, this will solve the problem. 

Thanks again!
